util/src/main/scala/org/ensime/util/io.scala

```diff
@@ -20,11 +32,20 @@ package object io {
      * endeavours to close everything afterward (even on failure).
      */
     def drain(in: InputStream): Unit =
-      try ByteStreams.copy(in, os)
-      finally {
+      try {
+        var data = Array.ofDim[Byte](16384) // size does affect perfomance
```

Comment:
where did you get the 16384 from? Link to the source would be really cool (also, we should probably do this everywhere if it is a known magical speed up)